modules/promotion/src/Entity/Promotion.php

```diff
@@ -243,9 +243,23 @@ public function apply(EntityInterface $entity) {
     $entity_type_id = $entity->getEntityTypeId();
     // @todo should whatever invokes this method be providing the context?
     $context = new Context(new ContextDefinition('entity:' . $entity_type_id), $entity);
+
+    // Execute each plugin, this is an AND operation.
+    // @todo support OR operations.
+    /** @var \Drupal\commerce\Plugin\Field\FieldType\PluginItem $item */
+    foreach ($this->get('conditions') as $item) {
```

Comment:
True. What if we wanted to check if a promotion "just applies" We'd want to have `applies` then. Should `apply` also then wrap the `applies` method?

Either way, no one should be loading a promotion directly and running these. There will be a service you pass an entity to, and that entity is processed.